cpp/tests/test_state_age_function.cpp

```diff
@@ -24,241 +24,382 @@
+    EXPECT_NEAR(wrapper_const.get_mean(), 1., 1e-14);
+
+    // Test if set_state_age_function works.
+    mio::ErlangDensity erl(2, 0.5);
+    wrapper_const.set_state_age_function(erl);
+    EXPECT_EQ(wrapper_const.get_parameter(), 2);
+    EXPECT_EQ(wrapper_const.get_scale(), 0.5);
+    EXPECT_NEAR(wrapper_const.get_support_max(dt), 14, 1e-14);
+    EXPECT_NEAR(wrapper_const.get_mean(dt), 1, 1e-14);
 
     // Reactive log output.
     mio::set_log_level(mio::LogLevel::warn);
 }
 
 TEST(TestStateAgeFunction, testSAFWrapperSettersAndGetters)
 {
-    // Test only for SmootherCosine since we have checked set_parameter and get_parameter
+    // Test getter and setter for Wrapper only for GammaSurvivalFunction since we have checked getter and setter
     // as well as get_support_max and get_mean for all derived classes in previous test.
 
-    ScalarType testvalue_before = 1.0;
-    ScalarType testvalue_after  = 2.0;
-    ScalarType dt               = 0.5;
-
-    mio::SmootherCosine smoothcos(testvalue_before);
-    mio::StateAgeFunctionWrapper wrapper(smoothcos);
-
-    EXPECT_EQ(wrapper.get_parameter(), testvalue_before);
-    EXPECT_EQ(wrapper.get_support_max(dt), testvalue_before);
-    // Use the formula for the mean of a SmootherCosine StateAgeFunction: 0.5 * parameter.
-    // See also the documentation of get_mean().
-    EXPECT_EQ(wrapper.get_mean(dt), testvalue_before * 0.5);
+    ScalarType dt = 0.5;
 
-    wrapper.set_parameter(testvalue_after);
-    EXPECT_EQ(wrapper.get_parameter(), testvalue_after);
-    EXPECT_EQ(wrapper.get_support_max(dt), testvalue_after);
-    EXPECT_EQ(wrapper.get_mean(dt), testvalue_after * 0.5);
+    mio::GammaSurvivalFunction gamma(1., 2., 3.);
+    mio::StateAgeFunctionWrapper wrapper(gamma);
+
+    EXPECT_EQ(wrapper.get_parameter(), 1.);
+    EXPECT_EQ(wrapper.get_location(), 2.);
+    EXPECT_EQ(wrapper.get_scale(), 3.);
+    EXPECT_NEAR(wrapper.get_support_max(dt), 71.5, 1e-14);
+    EXPECT_NEAR(wrapper.get_mean(), 5.0, 1e-14);
+
+    wrapper.set_parameter(4.5);
+    wrapper.set_location(0.5);
+    wrapper.set_scale(1.5);
+    EXPECT_EQ(wrapper.get_parameter(), 4.5);
+    EXPECT_EQ(wrapper.get_location(), 0.5);
+    EXPECT_EQ(wrapper.get_scale(), 1.5);
+    EXPECT_EQ(wrapper.get_support_max(dt), 50);
+    EXPECT_NEAR(wrapper.get_mean(), 7.25, 1e-4);
 }
 
 TEST(TestStateAgeFunction, testComparisonOperator)
 {
     // Check that StateAgeFunctions are only considered equal if they are of the same derived
     // class and have the same parameter.
-    mio::ExponentialDecay expdecay(0.5);
-    mio::ExponentialDecay expdecay2(0.5);
-    mio::ExponentialDecay expdecay3(1.0);
+    mio::GammaSurvivalFunction gamma(0.5, 0, 1);
+    mio::GammaSurvivalFunction gamma2(0.5, 0, 1);
+    mio::GammaSurvivalFunction gamma3(2, 0, 1);
+    mio::GammaSurvivalFunction gamma4(0.5, 1.5, 1);
+    mio::GammaSurvivalFunction gamma5(0.5, 0, 2);
     mio::SmootherCosine smoothcos(0.5);
```

Comment:
```suggestion
    mio::SmootherCosine smoothcos(0.5,0,1);
```
Just to make it more obvious that all parameters are the same (although these are the default values)

Reply:
SmootherCosine does not implement the scale parameter. Thats why i would keep it as before.

Reply:
Or use the LognormSF here?